### PR TITLE
added authorization requirement to endpoints

### PR DESCRIPTION
**Pull request by acn-sbuad**
## Description
- Added requirement of scope or accesstoken to relevant endpoints
- Implemented middleware for retrieving org info from user claims or access token issuer 
- Updated test project to reflect changed requirements
- Temp scope name set up, but not registered in maskinporten. Open to update after conversations with Maskinporten next week. 

## Related Issue(s)
- #8 

## Verification
- [x] **Your** code builds clean without any errors or warnings
- [x] Manual testing done (required)
- [x] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [x] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

File: src/Altinn.Notifications/Controllers/EmailNotificationOrdersController.cs [hunk 1]
```diff
@@ -60,7 +61,7 @@ public async Task<ActionResult<OrderIdExt>> Post(EmailNotificationOrderRequestEx
             return ValidationProblem(ModelState);
         }
 
-        string? creator = User.GetOrg();
+        string? creator = HttpContext.Items["Org"] as string;
```
Comment (SandGrainOne) [inline src/Altinn.Notifications/Controllers/EmailNotificationOrdersController.cs:64]: Is this defined in both cases? I wish we had something less magic, but we can save that for later.

File: src/Altinn.Notifications/Configuration/AuthorizationConstants.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    {
+        /// <summary>
+        /// Id for the policy requiring create scope or access platform access token
+        /// </summary>
+        public const string POLICY_CREATE_SCOPE_OR_PLATFORM_ACCESS = "CreateScopeOrPlatfomAccessToken";
+
+        /// <summary>
+        /// Scope for allowing access to creating notifications
+        /// </summary>
+        public const string SCOPE_NOTIFICATIONS_CREATE = "altinn:notifications.create";
```
Comment (SandGrainOne) [inline src/Altinn.Notifications/Configuration/AuthorizationConstants.cs:16]: We talked about using `altinn:serviceowner/notifications.create`, but I'm not sure.
Should we hold the door open for access by other parties?

File: src/Altinn.Notifications/Controllers/OrdersController.cs [hunk 3]
```diff
@@ -69,7 +69,7 @@ public async Task<ActionResult<NotificationOrderExt>> GetById([FromRoute] Guid i
     [SwaggerResponse(200, "The list of notification orders matching the provided senders ref was retrieved successfully", typeof(NotificationOrderListExt))]
     public async Task<ActionResult<NotificationOrderListExt>> GetBySendersRef([FromQuery, BindRequired] string sendersReference)
     {
-        string? expectedCreator = User.GetOrg();
+        string? expectedCreator = HttpContext.Items["Org"] as string;
```
[the remark this answers is not in the document]
Comment (acn-sbuad) [reply src/Altinn.Notifications/Controllers/OrdersController.cs:72]: created an extension method for http context for retrieval.

File: src/Altinn.Notifications/Controllers/OrdersController.cs
Comment (SandGrainOne) [inline src/Altinn.Notifications/Controllers/OrdersController.cs:1]: Should the endpoints here also require the same Policy?

Comment (acn-sbuad) [reply src/Altinn.Notifications/Controllers/OrdersController.cs:1]: Dang, yes they should. Fixing now. :)

File: test/Altinn.Notifications.IntegrationTests/Altinn.Notifications.IntegrationTests.csproj [hunk 4]
```diff
@@ -8,6 +8,10 @@
     <IsPackable>false</IsPackable>
   </PropertyGroup>
 
+  <ItemGroup>
```
Comment (SandGrainOne) [inline test/Altinn.Notifications.IntegrationTests/Altinn.Notifications.IntegrationTests.csproj:11]: Autogenerated noise.

**Review by SandGrainOne — APPROVED** [review]

**Pull request merged**